ui/src/app/features/application-header/application-header.component.html

```diff
@@ -15,15 +15,15 @@
   -->
 
 <div class="application-header">
-  <app-spinner *ngIf="loading$ | async"></app-spinner>
+  <app-spinner *ngIf="isLoading$ | async"></app-spinner>
   <mat-card class="application-header_left-coll">
     <div *ngIf="app$ | async as app">
       <h3 class="title">{{ 'applications.header.title' | translate }}</h3>
       <div class="name-wrapper">
         <div class="name-item" matTooltip="{{ app.name }}" [matTooltipDisabled]="app.name.length <= maxHeaderLinkLength">
           {{ app.name | truncate: maxHeaderLinkLength }}
         </div>
-        <div *ngIf="userRole$ | async as userRole">
+        <div *ngIf="!(isLoading$ | async) && userRole$ | async as userRole">
```

Comment:
need to refactor this component to container and presentational components in future to avoid such things

Reply:
Yes, you are right, I will refactor those components

Reply:
I meant to do it in the future, but thanks for updating now. Looks much better